app/controllers/todos_controller.rb

```diff
@@ -1,34 +1,48 @@
 class TodosController < ApplicationController
+  before_action :set_todo, only: [:update, :destroy]
+
   def index
     todos = Todo.where(username: params[:username])
     render json: { todos: todos }
   end
 
   def create
-    todo = Todo.new todo_params.merge(username: params[:username])
+    todo = Todo.new(todo_params.merge(username: params[:username]))
     if todo.save
       render json: { todo: todo }
     else
-      render status: 422, json: { errors: todo.errors.full_messages }
+      render json: { errors: @todo.errors.full_messages },
+             status: 422
     end
   end
 
   def update
-    todo = Todo.find_by(id: params[:id], username: params[:username])
-    if todo.present?
-      if todo.update todo_params
-        render json: { todo: todo }
-      else
-        render status: 422, json: { errors: todo.errors.full_messages }
-      end
+    if @todo.update(todo_params)
+      render json: { todo: @todo }
     else
-      render status: 404, json: { errors: ['Not found'] }
+      render json: { errors: @todo.errors.full_messages },
+             status: 422
     end
   end
 
+  def destroy
+    @todo.destroy
+    render json: {}
+  end
+
   private
 
   def todo_params
-    params.require(:todo).permit(:name)
+    params.require(:todo).permit(:name, :completed)
+  end
+
+  def set_todo
+    todo = Todo.find_by(id: params[:id], username: params[:username])
```

Comment:
I can also change this to assignment in the conditional but.. I didn't like the look of that.